gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/kmeans/KMeans.java

```diff
@@ -0,0 +1,147 @@
+ * of the vertex that are used for the clustering as input. Adds the clusterId, together with the
+ * cluster coordinates to the properties of the vertex. The datatype of the properties can be any numeric
+ * value. Returns the logical graph with modified vertex properties.
+ *
+ * @param <G>  The graph head type.
+ * @param <V>  The vertex type.
+ * @param <E>  The edge type.
+ * @param <LG> The type of the graph.
+ * @param <GC> The type of the graph collection.
+ */
+public class KMeans<G extends GraphHead, V extends Vertex, E extends Edge,
+  LG extends BaseGraph<G, V, E, LG, GC>, GC extends BaseGraphCollection<G, V, E, LG, GC>> implements
+  UnaryBaseGraphToBaseGraphOperator<LG> {
+
+  /**
+   * Number of iterations
+   */
+  private final int iterations;
+
+  /**
+   * Amount of clusters
+   */
+  private final int centroids;
+
+  /**ng
+   * Name of the first spatial property used for the clustering
+   */
+  private final String firstPropertyName;
+
+  /**
+   * Name of the second spatial property used for the clustering
+   */
+  private final String secondPropertyName;
+
+  /**
+   * Constructor to create an instance of KMeans
+   *
+   * @param iterations        Number of iterations, e.g., 20
+   * @param centroids         Amount of centroids that are determined by the algorithm
+   * @param propertyNameOne   First spatial property name of the vertices
+   * @param propertyNameTwo   Second spatial property name of the vertices
+   */
+  public KMeans(int iterations, int centroids, String propertyNameOne, String propertyNameTwo) {
+    Objects.requireNonNull(propertyNameOne);
+    Objects.requireNonNull(propertyNameTwo);
+    this.iterations = iterations;
+    this.centroids = centroids;
+    this.firstPropertyName = propertyNameOne;
+    this.secondPropertyName = propertyNameTwo;
+  }
+
+  @Override
+  public LG execute(LG logicalGraph) {
+
+    final String lat = this.firstPropertyName;
+    final String lon = this.secondPropertyName;
+
+    DataSet<V> spatialVertices =
+      logicalGraph.getVertices().filter(v -> v.hasProperty(lat) && v.hasProperty(lon));
+
+    DataSet<Point> points = spatialVertices.map(v -> {
+      double latValue = ((Number) v.getPropertyValue(lat).getObject()).doubleValue();
+      double lonValue = ((Number) v.getPropertyValue(lon).getObject()).doubleValue();
+      return new Point(latValue, lonValue);
+    });
+
+    DataSet<Tuple2<Long, Point>> indexingPoints = DataSetUtils.zipWithIndex(points.first(centroids));
+    DataSet<Centroid> firstCentroids =
+      indexingPoints.map(t -> new Centroid(Math.toIntExact(t.f0), t.f1.getLat(), t.f1.getLon()));
+
+    IterativeDataSet<Centroid> loop = firstCentroids.iterate(iterations);
+
+    DataSet<Centroid> newCentroids = points
+
+      //Assigns a centroid to every vertex
+        .map(new SelectNearestCenter()).withBroadcastSet(loop, "centroids")
+      /*
+      Adds a CountAppender to every Mapping and changes first value from centroidObject to
```

Comment:
Shorten to a single line comment, e.g. `// Add value 1 to prepare for grouping`

Reply:
Done